### DISCUSSION TITLE
metadata after including .qmd files in other qmd files

### DISCUSSION THREAD

**Discussion by kris-brown** [quarto-dev/quarto-cli#2959] [Q&A] [answered]
Hi, when I include a file `Y.qmd` (with title "Y") via `{{< include Y.qmd >}}` into file `X.qmd` with title "X" (where title is given in the metadata at the top of the file) the title of `X.qmd` is rendered as "Y", as its metadata seems to be drawn from the *final* "--- [metadata yaml] ---" block seen in the file, after pasting in the contents of `Y.qmd` where the include statement was. A workaround is to just put the metadata of `X.qmd` at the *bottom* of that file, but is this expected behavior? It feels natural for the metadata for a .qmd file to come from the *first* "--- [metadata yaml]---" block that is seen.

**Comment by cscheid** [comment, marked as answer]
Yes, this is expected behavior. As described in the documentation, `include` works literally: it behaves as if the entire contents of the file were pasted at the site of the `include` shortcode. This means that if you include a file with metadata, you get multiple blocks, and _pandoc_ interprets multiple blocks in this way.

The recommended workaround is to not `include` a `.qmd` file with metadata, but rather to `include` a `.md` file without metadata.

**Reply by kris-brown** [reply]
Thanks! It was natural to factor `Y.qmd` into `Y.md` and `Y.qmd` and then `include` the `Y.md` into both `Y.qmd` and `X.qmd`.

However, when quarto renders the `Y.qmd`, it seems like it automatically generates a `Y.md` for some intermediate computation and then automatically cleans up (i.e. deletes) `Y.md`. So one is precluded from ever having the same file name with both `.md` and `.qmd` extensions - this is a little unfortunate just because that convention would allow me to make it transparent "this is really the same file, but just wrapped with some metadata". My workaround is to just name the markdown file `Y_.md` but if quarto has any customizability to change how that intermediate file is named, that would be great to know!

**Reply by cscheid** [reply]
Oof. I'm sorry you got bit by that assumption in quarto.

Unfortunately, that convention is tied pretty deeply with quarto internals and it will be hard for us to change in the near term. I would recommend you use files _starting_ with `_` for includes, since those are automatically excluded from most of our wildcard calculations.
